docs/usage.md

```diff
@@ -30,74 +36,92 @@ For more specific integration options, please check out the [integration documen
 
 ## Understanding the output format
 
-TODO
+The output format is designed for you to use the generated models in further contexts. It might be part of a larger code generation such as AsyncAPI templates. This means that you can glue multiple models together into one large file, or split it out as you see fit.
 
 ## Generate models from AsyncAPI documents
 
-When providing an AsyncAPI document, the library iterates the entire document and generate models for all defined messages. If any other kind of iteration is wanted, feel free to create a [feature request](https://github.com/asyncapi/modelina/issues/new?assignees=&labels=enhancement&template=enhancement.md).
+When providing an AsyncAPI document, the library iterates the entire document and generate models for all defined message payloads. The message payloads are interpreted based on the schema format. 
+
+For JSON Schema it follows the [JSON Schema input processing](#generate-models-from-json-schema-documents).
 
 There are two ways to generate models for an AsyncAPI document.
 
-- [Generate from a parsed AsyncAPI document](../examples/asyncapi-from-parser)
-- [Generate from a pure JS object](../examples/asyncapi-from-object)
+- [Generate from a parsed AsyncAPI 2.x document](../examples/asyncapi-from-parser)
+- [Generate from an AsyncAPI 2.x JS object](../examples/asyncapi-from-object)
 
 The library expects the `asyncapi` property for the document to be set in order to understand the input format.
 
-The message payloads, since it is a JSON Schema variant, is [interpreted as a such](./interpretation_of_JSON_Schema.md).
+### Limitations and Compatibility
+These are the current known limitation of the AsyncAPI input.
+
+#### Message Schema formats
+Currently, only a limited number of schema formats are supported and we currently rely on the [AsyncAPI parser](github.com/asyncapi/parser-js/) to handle those schema formats and convert them into [AsyncAPI Schema format](https://github.com/asyncapi/parser-js/#custom-message-parsers). At some point in the future, Modelina will support all native schema formats, so it does not matter which standard you use to define the message payloads, you will be able to generate models from it.
```

Comment:
If the parser will not be able to support the given formats then the Modelina itself should not either. It seems to me that this paragraph is badly formatted, because even if Modelina will support e.g `xml` the parser still has to convert it to JSON first, or do you have a plan to do it in Modelina side?

Reply:
Think https://github.com/asyncapi/modelina/pull/857/files#r968490085 answers is right? 🤔